test/unit/test_visual_recognition_v4.py

```diff
@@ -12,824 +13,1199 @@
+# Test Class for train
+#-----------------------------------------------------------------------------
+class TestTrain():
+
+    #--------------------------------------------------------
+    # Test 1: Send fake data and check response
+    #--------------------------------------------------------
     @responses.activate
-    def test_train(self):
-        endpoint = '/v4/collections/{0}/train'.format('collection_id')
+    def test_train_response(self):
+        body = self.construct_full_body()
+        response = fake_response_Collection_json
+        send_request(self, body, response)
+        assert len(responses.calls) == 1
+
+    #--------------------------------------------------------
+    # Test 2: Send only required fake data and check response
+    #--------------------------------------------------------
+    @responses.activate
+    def test_train_required_response(self):
+        # Check response with required params
+        body = self.construct_required_body()
+        response = fake_response_Collection_json
+        send_request(self, body, response)
+        assert len(responses.calls) == 1
+
+    #--------------------------------------------------------
+    # Test 3: Send empty data and check response
+    #--------------------------------------------------------
+    @responses.activate
+    def test_train_empty(self):
+        check_empty_required_params(self, fake_response_Collection_json)
+        check_missing_required_params(self)
+        assert len(responses.calls) == 0
+
+    #-----------
+    #- Helpers -
+    #-----------
+    def make_url(self, body):
+        endpoint = '/v4/collections/{0}/train'.format(body['collection_id'])
         url = '{0}{1}'.format(base_url, endpoint)
-        response = {
-            "collection_id": "collection_id",
-            "training_status": {
-                "objects": {
-                    "in_progress": "true",
-                    "data_changed": "true",
-                    "ready": "true",
-                    "latest_failed": "true",
-                    "description": "description"
-                }
-            },
-            "created": "2000-01-23T04:56:07.000+00:00",
-            "name": "name",
-            "description": "description",
-            "image_count": 0,
-            "updated": "2000-01-23T04:56:07.000+00:00"
-        }
+        return url
+
+    def add_mock_response(self, url, response):
         responses.add(responses.POST,
                       url,
                       body=json.dumps(response),
                       status=202,
                       content_type='application/json')
 
-        authenticator = IAMAuthenticator('bogusapikey')
-        service = ibm_watson.VisualRecognitionV4('YYYY-MM-DD',
-                                                 authenticator=authenticator)
+    def call_service(self, body):
+        service = VisualRecognitionV4(authenticator=NoAuthAuthenticator(),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['VisualRecognitionV4' may be undefined, or defined from star imports: ibm_watson.visual_recognition_v4](https://app.codacy.com/app/watson-developer-cloud/python-sdk/pullRequest?prid=4546453)